### DISCUSSION TITLE
Refining ARKit ARCore

### DISCUSSION THREAD

**Discussion by FreakTheMighty** [openMVG/openMVG#2149] [Ideas]
Hi all,

I've seen some interest in using AR captures as a starting point for SfM reconstructions and I'm interested in this too. I suspect this will have growing relevance as mobile AR proliferates and I'm curious to hear what other's goals are and whether there is opportunity to collaborate? 

For my part, I am interested in using data captured through ARKit, including intrinsics, extrinsics, point clouds, along with distance constraints between known points, and calculating a more accurate SfM solution then what can be done in realtime on the phone. 

----
Some questions

* Can point cloud + pose data be used to reduce the number of images that are compared during image matching to improve performance?
* Can the initial guess for camera poses be used as a starting place for OpenMVG and be refined to be more accurate?
* Can distance constraints between known points be used to mitigate some of the scale/drift that is commonly found in mobile AR maps?
* Is there interest in the community to build tools within OpenMVG for this specific use case or to spin up another repository that explores pipelines that explore this use case? I can imagine a collaborative effort involving a sample iOS app + an OpenMVG pipeline.

https://github.com/openMVG/openMVG/issues/1350

@AlbertoLanaro
@masterchief1s1k
@marcontex8
@RishabhBajaj25
@num-doc

**Comment by num-doc** [comment]
arkit has its own set of coordinate system. Unfortunately, Apple only provides the calling interface, and the actual 3D point cloud data obtained by arkit cannot be directly used for 3D reconstruction

**Reply by FreakTheMighty** [reply]
I'm not suggesting the point cloud be used directly. I think it could be used to guess which image pairs will likely have overlapping features. If points show up in both camera frustums, than we can assume its worth performing an image match on the pair.
